functests/dpdk/dpdk.go

```diff
@@ -581,6 +581,68 @@ func ValidateSriovPolicy() (bool, error) {
 	return false, nil
 }
 
+func findPodByNodeFromList(nodeName string, podList []corev1.Pod) (corev1.Pod, error) {
+	for _, pod := range podList {
+		if pod.Spec.NodeName == nodeName {
+			return pod, nil
+		}
+	}
+
+	return corev1.Pod{}, fmt.Errorf("failed to find pod on node %s", nodeName)
+}
+
+// FindDpdkSriovDevice will search for the default gateway interface to be used as the PF
+func FindDpdkSriovDevice(enabledNodes *sriovcluster.EnabledNodes, nodeName string) (*sriovv1.InterfaceExt, error) {
+	s, ok := enabledNodes.States[nodeName]
+	if !ok {
+		return nil, fmt.Errorf("Node %s not found", nodeName)
+	}
+
+	podList := &corev1.PodList{}
+	err := client.Client.List(context.Background(), podList, &goclient.ListOptions{Namespace: SRIOV_OPERATOR_NAMESPACE,
+		LabelSelector: labels.SelectorFromSet(labels.Set{"app": "sriov-network-config-daemon"})})
```

Comment:
Can't you just use FieldsSelector here and avoid the findPodBy..?

Reply:
+1